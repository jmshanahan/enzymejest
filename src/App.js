import React from 'react';
import logo from './logo.svg';
import './App.css';
const Test = () => <div>Testing</div>;

function fetchBoxes() {
  return new Promise(resolve => {
    setTimeout(() => {
      resolve(['First Box', 'Second Box', 'Third Box']);
    }, 1000);
  });
}
class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      on: false,
      input: '',
      mainColor: 'blue',
      boxes: [],
      lifeCycle: ''
    };
  }
  componentDidMount() {
    this.setState({ lifeCycle: 'componentDidMountut' });

    fetchBoxes().then(boxes => this.setState({ boxes }));
  }
  componentWillReceiveProps() {
    this.setState({ lifeCycle: 'componentWillReceiveProps' });
  }
  render() {
    return (
      <div className="App">
        <h1 className="App-title">Welcome to React</h1>
        <h3 className={this.state.mainColor}>Everyone is Welcome</h3>
        <p className="lifeCycle">{this.state.lifeCycle}</p>
        <Test />
        <p className="button-state">{this.state.on ? 'Yes!' : 'No!'}</p>
        <button onClick={() => this.setState({ on: true })}>Click</button>
        <h2>{this.state.input}</h2>
        <input
          onChange={e => this.setState({ input: e.currentTarget.value })}
          type="text"
        />
      </div>
    );
  }
}

export default App;
