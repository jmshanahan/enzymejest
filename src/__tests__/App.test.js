import React from 'react';
import ReactDOM from 'react-dom';
import '../../enzyme.config';
import App from '../App';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('App shallow rendering', () => {
  it('should render app', () => {
    const wrapper = shallow(<App />, {
      context: {},
      disableLifecycleMethods: true
    });
    expect(wrapper.find('h1').text()).toBe('Welcome to React');
    // console.log(wrapper.debug());
  });
  it('matches the snapshot', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('on button click changes p text', () => {
    const wrapper = shallow(<App />);
    const button = wrapper.find('button');
    expect(wrapper.find('.button-state').text()).toBe('No!');
    button.simulate('click');
    expect(wrapper.find('.button-state').text()).toBe('Yes!');
  });
  it('on input change, title changes text', () => {
    const wrapper = shallow(<App />);
    const input = wrapper.find('input');
    expect(wrapper.find('h2').text()).toBe('');
    input.simulate('change', { currentTarget: { value: 'Tyler' } });
    expect(wrapper.find('h2').text()).toBe('Tyler');
  });
});
describe('App mount rendering ', () => {
  it('should render app', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find('h1').text()).toBe('Welcome to React');
    wrapper.unmount();
  });
  it('matches the snapshot', () => {
    const wrapper = mount(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
