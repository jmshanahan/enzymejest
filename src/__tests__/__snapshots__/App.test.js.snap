// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App mount rendering  matches the snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
    <Test>
      <div>
        Testing
      </div>
    </Test>
    <p
      className="button-state"
    >
      No!
    </p>
    <button
      onClick={[Function]}
    >
      Click
    </button>
    <h2 />
    <input
      onChange={[Function]}
      type="text"
    />
  </div>
</App>
`;

exports[`App shallow rendering matches the snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="App-title"
  >
    Welcome to React
  </h1>
  <Test />
  <p
    className="button-state"
  >
    No!
  </p>
  <button
    onClick={[Function]}
  >
    Click
  </button>
  <h2 />
  <input
    onChange={[Function]}
    type="text"
  />
</div>
`;
